### PR TITLE
Add support for new source scrape auth fields.

### PR DESCRIPTION
**Pull request by alistairjevans**

### CODE REVIEW

File: internal/provider/resource_source.go [hunk 1]
```diff
@@ -165,6 +165,28 @@ var sourceSchema = map[string]*schema.Schema{
...
+		Description: "An array of request headers, each containing `name` and `value` fields.",
+		Type:        schema.TypeList,
+		Optional:    true,
+		Elem: &schema.Schema{
+			Type: schema.TypeMap,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
```
Comment (PetrHeinz) [inline internal/provider/resource_source.go:178]: I can see the headers are loaded as a map, but we only expect `value` and `name`. Users had issues with that in Uptime provider, intuitively filling the map as eg. `{"Bearer": "mytoken"}` which then failed on API in the apply phase.

Could you please add validation for the map and test it in the `resource_source_test.go`? 🙏 

We're doing that in Uptime:
Validation: https://github.com/BetterStackHQ/terraform-provider-better-uptime/blob/6e5c3466129935ee5b98f1ba163fa534d460dcc1/internal/provider/resource_monitor.go#L610
Testing: https://github.com/BetterStackHQ/terraform-provider-better-uptime/blob/6e5c3466129935ee5b98f1ba163fa534d460dcc1/internal/provider/resource_monitor_test.go#L222

File: internal/provider/resource_source_test.go [hunk 2]
```diff
@@ -203,7 +211,12 @@ func TestResourceSource(t *testing.T) {
 					resource.TestCheckResourceAttr("logtail_source.this", "ingesting_paused", "true"),
 					resource.TestCheckResourceAttr("logtail_source.this", "token", "generated_by_logtail"),
 					resource.TestCheckResourceAttr("logtail_source.this", "scrape_urls.#", "1"),
-					resource.TestCheckResourceAttr("logtail_source.this", "scrape_frequency_secs", "30"),
```
Comment (PetrHeinz) [inline internal/provider/resource_source_test.go:206]: Probably removed by accident?

**Review by PetrHeinz — COMMENTED** [review]
Thanks @alistairjevans 🙌 

Looks great! There are couple of things we might do to ensure it doesn't break in the future - plan phase validation and end to end testing.

Could you please add the new possibility into `examples`? You can either add it to the `examples/advanced`, or create your own dedicated example. If you do the latter, please add it to https://github.com/BetterStackHQ/terraform-provider-logtail/blob/main/.github/workflows/test.yml#L32 so it too gets automatically tested 🙏 

Examples using new features should have their required version set to the newly released one, as they cannot be run with previous versions (see https://github.com/BetterStackHQ/terraform-provider-better-uptime/commit/14b3b863da2747194b09f5cf8ba3b86d4dfcbe53 for an example).

**Comment by alistairjevans** [comment]
Thanks for the pointers @PetrHeinz; I went for a completely new example, and added that extra validation. 👍

File: examples/scrape/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+output "logtail_source_token" {
+  value = logtail_source.this.token
+}
```
Comment (PetrHeinz) [inline examples/scrape/outputs.tf]: 💄 
```suggestion
}

```

File: examples/scrape/main.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  scrape_urls           = ["https://myserver.example.com/metrics"]
+  scrape_frequency_secs = 30
+  scrape_request_headers = [
+    {
+      name  = "User-Agent"
+      value = "My Scraper"
+    }
+  ]
+  scrape_request_basic_auth_user     = "foo"
+  scrape_request_basic_auth_password = "bah"
```
Comment (PetrHeinz) [inline examples/scrape/main.tf]: 💄 
```suggestion
  scrape_request_basic_auth_password = "bar"
```

File: internal/provider/resource_source_test.go [hunk 5]
```diff
@@ -194,6 +195,14 @@ func TestResourceSource(t *testing.T) {
...
 					scrape_urls      = ["http://localhost:9100/metrics"]
 					scrape_frequency_secs = 30
+					scrape_request_basic_auth_user = "user1"
+					scrape_request_basic_auth_password = "password1"
+					scrape_request_headers = [
+						{ 
+							name = "Authorization",
+							value = "Bearer foo"
+						}
+					]
```
Comment (PetrHeinz) [inline internal/provider/resource_source_test.go:205]: 💄 Since you've added comprehensive test of scraping below, I would revert these changes to keep the test simple - WDYT? 

Adding the rest of removals as suggestions for easy update if you decide to do so 👍 

```suggestion
```

File: internal/provider/resource_source_test.go [hunk 6]
```diff
@@ -203,7 +212,13 @@ func TestResourceSource(t *testing.T) {
 					resource.TestCheckResourceAttr("logtail_source.this", "ingesting_paused", "true"),
 					resource.TestCheckResourceAttr("logtail_source.this", "token", "generated_by_logtail"),
 					resource.TestCheckResourceAttr("logtail_source.this", "scrape_urls.#", "1"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_urls.0", "http://localhost:9100/metrics"),
```
Comment (PetrHeinz) [inline internal/provider/resource_source_test.go:215]:
```suggestion
```

File: internal/provider/resource_source_test.go [hunk 7]
```diff
@@ -203,7 +212,13 @@ func TestResourceSource(t *testing.T) {
...
 					resource.TestCheckResourceAttr("logtail_source.this", "scrape_frequency_secs", "30"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_request_headers.#", "1"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_request_headers.0.name", "Authorization"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_request_headers.0.value", "Bearer foo"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_request_basic_auth_user", "user1"),
+					resource.TestCheckResourceAttr("logtail_source.this", "scrape_request_basic_auth_password", "password1"),
```
Comment (PetrHeinz) [inline internal/provider/resource_source_test.go:221]:
```suggestion
```

File: internal/provider/resource_source_test.go [hunk 8]
```diff
@@ -218,6 +233,14 @@ func TestResourceSource(t *testing.T) {
...
 					scrape_urls      = ["http://localhost:9100/metrics"]
 					scrape_frequency_secs = 30
+					scrape_request_basic_auth_user = "user1"
+					scrape_request_basic_auth_password = "password1"
+					scrape_request_headers = [
+						{ 
+							name = "Authorization",
+							value = "Bearer foo"
+						}
+					]
```
Comment (PetrHeinz) [inline internal/provider/resource_source_test.go:243]:
```suggestion
```

**Review by PetrHeinz — APPROVED** [review]
Awesome @alistairjevans 🙌 

Thanks for the changes, good to go ✅  I have a couple nitpicks, everything added as a simple code suggestion.

File: internal/provider/resource_source_test.go (hunk 5 above)
Comment (alistairjevans) [reply internal/provider/resource_source_test.go:205]: I didn't intend to capture all the other scrape request attributes checks in the header tests. I'll simplify the header check tests to just check headers, and leave the existing test as a "set all the attributes" test.
